Opcodes/phisem.c

```diff
@@ -193,1106 +193,1106 @@ static MYFLT noise_tick(CSOUND *csound)
-    p->kloop = (int32_t)(p->h.insdshead->offtim * CS_EKR)
-             - (int32_t)(CS_EKR * *p->dettack);
-
-    p->outputs00       = FL(0.0);
-    p->outputs01       = FL(0.0);
-    p->outputs10       = FL(0.0);
-    p->outputs11       = FL(0.0);
-    p->outputs20       = FL(0.0);
-    p->outputs21       = FL(0.0);
-
-    p->totalEnergy     = FL(0.0);
-
-    p->center_freqs0   = p->res_freq0 = WUTR_CENTER_FREQ0;
-    p->center_freqs1   = p->res_freq1 = WUTR_CENTER_FREQ1;
-    p->center_freqs2   = p->res_freq2 = WUTR_CENTER_FREQ2;
-    p->num_objectsSave = p->num_objects = WUTR_NUM_SOURCES;
-    p->soundDecay      = WUTR_SOUND_DECAY;
-    p->systemDecay     = WUTR_SYSTEM_DECAY;
-    temp               = LOG(WUTR_NUM_SOURCES) * WUTR_GAIN / WUTR_NUM_SOURCES;
-    p->gains0          = p->gains1 = p->gains2 = temp;
-    p->coeffs01        = WUTR_RESON * WUTR_RESON;
-    p->coeffs00        = -WUTR_RESON * FL(2.0) *
-      COS(WUTR_CENTER_FREQ0 * CS_TPIDSR);
-    p->coeffs11        = WUTR_RESON * WUTR_RESON;
-    p->coeffs10        = -WUTR_RESON * FL(2.0) *
-      COS(WUTR_CENTER_FREQ1 * CS_TPIDSR);
-    p->coeffs21        = WUTR_RESON * WUTR_RESON;
-    p->coeffs20        = -WUTR_RESON * FL(2.0) *
-      COS(WUTR_CENTER_FREQ2 * CS_TPIDSR);
-                                /* Note On */
-    p->shakeEnergy     = *p->amp * (FL(1.)/csound->Get0dBFS(csound)) * MAX_SHAKE * FL(0.1);
-    p->shake_damp      = FL(0.0);
-    if (p->shakeEnergy > MAX_SHAKE) p->shakeEnergy = MAX_SHAKE;
-    p->shake_maxSave = FL(0.0);
-    p->num_objects = 10;        /* Bug fix by JPff 2014/08/27 */
-    p->finalZ0 = p->finalZ1 = p->finalZ2 = FL(0.0);
-    return OK;
+  MYFLT temp;
+
+  p->sndLevel = FL(0.0);
+  p->kloop = (int32_t)(p->h.insdshead->offtim * CS_EKR)
+    - (int32_t)(CS_EKR * *p->dettack);
+
+  p->outputs00       = FL(0.0);
+  p->outputs01       = FL(0.0);
+  p->outputs10       = FL(0.0);
+  p->outputs11       = FL(0.0);
+  p->outputs20       = FL(0.0);
+  p->outputs21       = FL(0.0);
+
+  p->totalEnergy     = FL(0.0);
+
+  p->center_freqs0   = p->res_freq0 = WUTR_CENTER_FREQ0;
+  p->center_freqs1   = p->res_freq1 = WUTR_CENTER_FREQ1;
+  p->center_freqs2   = p->res_freq2 = WUTR_CENTER_FREQ2;
+  p->num_objectsSave = p->num_objects = WUTR_NUM_SOURCES;
+  p->soundDecay      = WUTR_SOUND_DECAY;
+  p->systemDecay     = WUTR_SYSTEM_DECAY;
+  temp               = LOG(WUTR_NUM_SOURCES) * WUTR_GAIN / WUTR_NUM_SOURCES;
+  p->gains0          = p->gains1 = p->gains2 = temp;
+  p->coeffs01        = WUTR_RESON * WUTR_RESON;
+  p->coeffs00        = -WUTR_RESON * FL(2.0) *
+    COS(WUTR_CENTER_FREQ0 * CS_TPIDSR);
+  p->coeffs11        = WUTR_RESON * WUTR_RESON;
+  p->coeffs10        = -WUTR_RESON * FL(2.0) *
+    COS(WUTR_CENTER_FREQ1 * CS_TPIDSR);
+  p->coeffs21        = WUTR_RESON * WUTR_RESON;
+  p->coeffs20        = -WUTR_RESON * FL(2.0) *
+    COS(WUTR_CENTER_FREQ2 * CS_TPIDSR);
+  /* Note On */
+  p->shakeEnergy     = *p->amp * CS_DBFS_FLOAT * MAX_SHAKE * FL(0.1);
```

Comment:
Our other macros like CS_ONEDKSMPS and CS_ONEDKR use ONED for 1/x values. Seems like this macro should be renamed CS_ONEDDBFS_FLOAT. (Or CS_ONEDDBFS? Not sure what having _FLOAT differentiates)